### PR TITLE
Add cfg header

### PR DESCRIPTION
**Pull request by jontje**
As per title

### CODE REVIEW

File: include/abb_librws/rws_cfg.h [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+/**
+ * \brief Namespace for configuration domains.
+ */
+namespace cfg
+{
+
+/**
+ * \brief Namespace for types belonging to the motion (MOC) domain.
+ */
+namespace moc
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_cfg.h:57]: That's some deep namespacing here: `abb::rws::cfg::moc::`.

File: include/abb_librws/rws_cfg.h [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+   */
+  std::string name;
+
+  /**
+   * \brief The lower joint bound.
+   *
+   * The unit depends on the type of joint, i.e. [rad] for rotational and [m] for linear.
+   *
+   */
+  double lower_joint_bound;
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_cfg.h:75]: Not sure I understand how an "arm" can have a single "lower bound". I would have expected multiple, or `Arm` being called a `Joint`.

But I might not be up-to-speed on RWS nomenclature.

File: include/abb_librws/rws_cfg.h [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+  /**
+   * \brief The joint's kinematic axis number.
+   */
+  int kinematic_axis_number;
+
+  /**
+   * \brief The arm used by the joint.
+   */
+  std::string use_arm;
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_cfg.h:108]: Is `use_arm` what RWS uses? I would've expected `used_arm` (as in: the arm which is used).

File: include/abb_librws/rws_cfg.h [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+  /**
+   * \brief The robot used by the mechanical unit.
+   */
+  std::string use_robot;
+
+  /**
+   * \brief The singles used by the mechanical unit.
+   */
+  std::vector<std::string> use_singles;
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_cfg.h:134]: What do we store in this `vector`? Names associated with objects?

Could we use instances of actual types here instead?

File: include/abb_librws/rws_cfg.h [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+  /**
+   * \brief The mechanical units (without TCP) used by the group.
+   */
+  std::vector<std::string> mechanical_units;
+};
+
+/**
+ * \brief Representation of a 'Present Option' instance.
+ */
+struct PresentOption
```
Comment (gavanderhoorn) [inline include/abb_librws/rws_cfg.h:245]: Preset is different from *active* I gather?

**Review by gavanderhoorn — COMMENTED** [review]
Comments again.

File: include/abb_librws/rws_cfg.h (hunk 1 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:57]: Yes... I know 😅

File: include/abb_librws/rws_cfg.h (hunk 2 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:75]: It was also confusing for me when I dug down into the definitions, and I don't know the history behind the naming.

Anyway, the intention of the `rws_cfg` file is to have representations of how `RobotWare` structures its configurations, and I have tried to mirror this as much as possible.

File: include/abb_librws/rws_cfg.h (hunk 3 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:108]: It is what `RobotWare` uses in its configuration structure, and I think it is reasonable to have the same name here as is used in the real systems to minimize confusion.

File: include/abb_librws/rws_cfg.h (hunk 4 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:134]: Yes, it stores the names, and again, this is how the `RobotWare` configurations are structured.

I think it is more reasonable to leave the "parsing" up to actual use cases. This is manily to structure data on the same format as `RobotWare` does.

File: include/abb_librws/rws_cfg.h (hunk 5 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:245]: In this case, then _active_ and _present_ are the same.

My bad if my previously commenting has been confusing.

File: include/abb_librws/rws_cfg.h (hunk 2 above)
Comment (gavanderhoorn) [reply include/abb_librws/rws_cfg.h:75]: Ok. I assumed as much.

Is there any documentation or nomenclature document we could refer to here in a comment (perhaps at the top of the file)?

File: include/abb_librws/rws_cfg.h (hunk 3 above)
Comment (gavanderhoorn) [reply include/abb_librws/rws_cfg.h:108]: Yes, I agree on the use-whatever-robotware-has.

I'm just surprised to see this name.

**Review by gavanderhoorn — DISMISSED** [review]

File: include/abb_librws/rws_cfg.h (hunk 2 above)
Comment (jontje) [reply include/abb_librws/rws_cfg.h:84]: Added in 0c097cf

**Review by gavanderhoorn — APPROVED** [review]

**Comment by gavanderhoorn** [comment]
Let's wait for Travis and then merge.

**Comment by jontje** [comment]
Great, thanks for the review @gavanderhoorn 👍

**Pull request merged**